.github/workflows/baremetal-regression-suite.yml

```diff
@@ -0,0 +1,177 @@
+name: Baremetal Regression Suite
+
+on:
+  pull_request:
+    branches:
+      - "master"
+      - "candidate-*"
+
+  workflow_dispatch:
+
+jobs:
+  test-against-platform:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Setup JDK 11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11
+
+    - uses: "actions/setup-python@v2"
+      with:
+        python-version: "3.8"
+
+    - name: "Install dependencies"
+      run: |
+        set -xe
+        python -VV
+        python -m site
+        python -m pip install --upgrade pip setuptools wheel
+
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+
+    - name: Rebase
+      if: github.event_name == 'pull_request'
```

Comment:
does github provide a mechanism to save and re-use jobs? If so, it feels like we should reuse some of these